Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -854,7 +856,9 @@ validate personId phoneNumber = do
   altNoAttempt <- runInReplica $ QRegister.getAlternateNumberAttempts personId
   runRequestValidation validationCheck phoneNumber
   mobileNumberHash <- getDbHash phoneNumber.alternateNumber
-  duplicateCheck (QPerson.findByMobileNumberAndMerchant phoneNumber.mobileCountryCode mobileNumberHash person.merchantId) "Person with this mobile number already exists"
+  merchant <- CQM.findById person.merchantId >>= fromMaybeM (InvalidRequest "Merchant not found")
+  mbPerson <- QPerson.findByMobileNumberAndMerchant phoneNumber.mobileCountryCode mobileNumberHash person.merchantId
+  whenJust mbPerson $ DeleteDriverOnCheck.merchantAccessCheck merchant
```

Comment:
merchantAccessCheck should just return one boolean and then error should be thrown accordingly instead of this function itself throwing the error.